### ISSUE TITLE
src/pyglue/PyAllocationTransform.cpp:68:27: error: cast between incompatible function types

### ISSUE THREAD

**Issue by hobbes1069** [AcademySoftwareFoundation/OpenColorIO#719]
Fedora Rawhide, GCC 9.0.1, Python 3.7

```
[ 71%] Building CXX object src/pyglue/CMakeFiles/PyOpenColorIO.dir/PyAllocationTransform.cpp.o
cd /builddir/build/BUILD/OpenColorIO-1.1.1/build/src/pyglue && /usr/bin/c++  -DPYOCIO_NAME=PyOpenColorIO -DPyOpenColorIO_EXPORTS -DUSE_SSE -I/builddir/build/BUILD/OpenColorIO-1.1.1/export -I/builddir/build/BUILD/OpenColorIO-1.1.1/build/export -I/builddir/build/BUILD/OpenColorIO-1.1.1/build/src/pyglue -I/usr/include/python3.7m  -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wall -Wextra -Wshadow -Wconversion -Wcast-qual -Wformat=2 -msse2 -Wno-strict-aliasing -Wno-missing-field-initializers -Werror -O2 -g -DNDEBUG -fPIC   -o CMakeFiles/PyOpenColorIO.dir/PyAllocationTransform.cpp.o -c /builddir/build/BUILD/OpenColorIO-1.1.1/src/pyglue/PyAllocationTransform.cpp
BUILDSTDERR: /builddir/build/BUILD/OpenColorIO-1.1.1/src/pyglue/PyAllocationTransform.cpp:68:27: error: cast between incompatible function types from 'PyObject* (*)(PyObject*)' {aka '_object* (*)(_object*)'} to 'PyCFunction' {aka '_object* (*)(_object*, _object*)'} [-Werror=cast-function-type]
BUILDSTDERR:    68 |             (PyCFunction) PyOCIO_AllocationTransform_getAllocation, METH_NOARGS, ALLOCATIONTRANSFORM_GETALLOCATION__DOC__ },
BUILDSTDERR:       |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/OpenColorIO-1.1.1/src/pyglue/PyAllocationTransform.cpp:72:27: error: cast between incompatible function types from 'PyObject* (*)(PyObject*)' {aka '_object* (*)(_object*)'} to 'PyCFunction' {aka '_object* (*)(_object*, _object*)'} [-Werror=cast-function-type]
BUILDSTDERR:    72 |             (PyCFunction) PyOCIO_AllocationTransform_getNumVars, METH_VARARGS, ALLOCATIONTRANSFORM_GETNUMVARS__DOC__ },
BUILDSTDERR:       |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/OpenColorIO-1.1.1/src/pyglue/PyAllocationTransform.cpp:74:27: error: cast between incompatible function types from 'PyObject* (*)(PyObject*)' {aka '_object* (*)(_object*)'} to 'PyCFunction' {aka '_object* (*)(_object*, _object*)'} [-Werror=cast-function-type]
BUILDSTDERR:    74 |             (PyCFunction) PyOCIO_AllocationTransform_getVars, METH_NOARGS, ALLOCATIONTRANSFORM_GETVARS__DOC__ },
BUILDSTDERR:       |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: /builddir/build/BUILD/OpenColorIO-1.1.1/src/pyglue/PyAllocationTransform.cpp:56:20: error: 'PyObject* OpenColorIO::v1::{anonymous}::PyOCIO_AllocationTransform_equals(PyObject*, PyObject*)' declared 'static' but never defined [-Werror=unused-function]
BUILDSTDERR:    56 |         PyObject * PyOCIO_AllocationTransform_equals(PyObject * self,  PyObject * args);
BUILDSTDERR:       |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
BUILDSTDERR: cc1plus: all warnings being treated as errors
```

**Comment by michdolan** [comment]
OCIO does not currently support Python 3; can you try with 2.7?

**Comment by hobbes1069** [comment]
Nope... Fedora Rawhide (31) no longer allows python2 only packages as Python 2 is EOL at the end of the year. I guess I'll just disable the python package. I don't think there were any consumers of it on Fedora anyway.

**Comment by lgritz** [comment]
Aside: we are currently trying to maintain the hope of not needing any compatibility-breaking or feature-changing releases until v2 is completely finished with its full original planned feature set, more than a year from now. The need for studios to have a python3-compatible OCIO in time for VFX Platform 2020 (not to mention Python 2's EOL) might be the immovable post that clashes with our unstoppable v2 cannonball.

**Comment by matthewdcong** [comment]
I tried this on my end with:

- OSX 10.14, Python 3.7, clang
- OSX 10.14, Python 3.7, gcc-9

It worked in both cases for me on a simple Python 3 example of applying a processor to a pixel value. Thus, I'm optimistic regarding Python 3 support in RB-1.1.

The warnings that you're seeing have been addressed since the OCIO 1.1.1 release. PR #838 addresses the first set of warnings regarding casts between incompatible function types and 1c62465 addresses the second unused function error.
